include/zephyr/lorawan/lorawan.h

```diff
@@ -166,10 +166,12 @@ struct lorawan_downlink_cb {
 	 * @param snr Signal to Noise ratio in dBm
 	 * @param len Length of data received, will be 0 for ACKs
 	 * @param data Data received, will be NULL for ACKs
+	 * @param time Time update has been received
 	 */
 	void (*cb)(uint8_t port, bool data_pending,
 		   int16_t rssi, int8_t snr,
-		   uint8_t len, const uint8_t *data);
+		   uint8_t len, const uint8_t *data,
+		   bool time);
```

Comment:
Given we are breaking API compatibility here anyway, my preference would be to change `bool data_pending` to `uint8_t  flags`, together with a definition like:
```
enum lorwawan_dl_flags {
  LORAWAN_DATA_PENDING = BIT(0),
  LORAWAN_TIME_UPDATED = BIT(1),
};
```
This would let us add additional booleans (OTA update related?) without continuously breaking the compatibility.